apps/chat/src/components/AppsEditor/GeneralInfoView/GeneralInfoView.tsx

```diff
@@ -56,11 +67,38 @@ export const GeneralInfoView: React.FC<Props> = ({
     ),
   });
 
+  const { t } = useTranslation(Translation.Chat);
+
   const formData = methods.watch();
 
+  const screenState = useScreenState();
+
+  const [previewMode, setPreviewMode] = useState<PreviewMode>(() => {
+    return screenState === ScreenState.MD
+      ? PreviewMode.closed
+      : PreviewMode.half;
+  });
+
+  useEffect(() => {
+    if (screenState === ScreenState.MD) {
+      setPreviewMode(PreviewMode.closed);
+    } else {
+      setPreviewMode(PreviewMode.half);
+    }
```

Comment:
`screenState` can be `SM`(mobile)
We shouldn't reset possible modes during resize (e.g. SM <-> MD and `full`)
```suggestion
    if (screenState > ScreenState.MD && previewMode !== PreviewMode.half) {
      setPreviewMode(PreviewMode.half);
    } else if(previewMode === PreviewMode.half) {
      setPreviewMode(PreviewMode.closed);
    }
```

Reply:
The only thing you need to add is a condition inside else, because else is not the only alternative to screenState <= ScreenState.MD. There may be intermediate states that cause opening and closing glitches.

useEffect(() => {
    if (screenState > ScreenState.MD && previewMode !== PreviewMode.half) {
      setPreviewMode(PreviewMode.half);
    } else if (
      screenState <= ScreenState.MD &&
      previewMode === PreviewMode.half
    ) {
      setPreviewMode(PreviewMode.closed);
    }
  }, [screenState, previewMode]);
  
  This option works without glitches.